File: API/WorkoutTracker.API/Models/ExecutedWorkoutSummaryDTO.cs
using System;

namespace WorkoutTracker.API.Models
{
    public record ExecutedWorkoutSummaryDTO(Guid Id, DateTime CreatedDateTime, DateTime? ModifiedDateTime, string Name,
        Guid WorkoutPublicId,
        DateTime? StartDateTime,
        DateTime? EndDateTime,
        string Journal) 
        : NamedEntityDTO(Id, CreatedDateTime, ModifiedDateTime, Name);
}

File: workout-tracker/src/app/workouts/models/executed-workout-summary-dto.ts
import { NamedEntityDTO } from "./named-entity-dto";

export class ExecutedWorkoutSummaryDTO extends NamedEntityDTO {
  public workoutPublicId: string = ''; //GUID
  public startDateTime: Date = new Date();
  public endDateTime: Date = new Date();
  public journal: string | null = null;
}
